src/event-stream/event-server.ts

```diff
@@ -190,6 +207,44 @@ async function handleClientMessage(msg: CoreNodeBlockMessage, db: DataStore): Pr
           value: hexToBuffer(event.contract_event.raw_value),
         };
         dbTx.contractLogEvents.push(entry);
+        if (
+          event.contract_event.topic === printTopic &&
+          event.contract_event.contract_identifier === bnsContractIdentifier
+        ) {
+          const functionName = getFunctionName(event.txid, parsedMsg.parsed_transactions);
+          if (nameFunctions.includes(functionName)) {
+            const attachment = parseNameRawValue(event.contract_event.raw_value);
+            const attachmentValue = await parseContentHash(attachment.attachment.hash);
+
+            const names: DbBNSName = {
+              name: attachment.attachment.metadata.name,
+              namespace_id: attachment.attachment.metadata.namespace,
+              address: addressToString(attachment.attachment.metadata.tx_sender),
+              expire_block: 0, // FIXME:
+              registered_at: parsedMsg.block_height,
+              zonefile_hash: attachment.attachment.hash,
+              zonefile: attachmentValue,
+              latest: true,
+              tx_id: event.txid,
+              status: attachment.attachment.metadata.op,
+              index_block_hash: parsedMsg.index_block_hash,
+              canonical: true,
+            };
+            console.log('update names ', JSON.stringify(names));
+            await db.updateNames(names);
```

Comment:
Performing database updates here can cause data integrity issues. This should instead be something like `dbData.names.push(name)`, then the `await db.update(dbData)` function called at the end of this function should be performing these updates within its postgres transaction.

Reply:
Updated this as suggested.